workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestList/hooks/useFilters.tsx

```diff
@@ -13,13 +13,40 @@ import getActivityPeriodFilterTerm from '../utils/getActivityPeriodFilterTerm';
 
 export default function useFilters(
 	openRequestFilter: boolean,
+	urlParams: URLSearchParams,
 	isChannel?: boolean
 ) {
-	const [filters, setFilters] = useState(
-		(JSON.parse(
-			sessionStorage.getItem('requestFilters')!
-		) as typeof INITIAL_FILTER) || INITIAL_FILTER
-	);
+	const [filters, setFilters] = useState(() => {
+		const initialFilter: typeof INITIAL_FILTER = JSON.parse(
+			JSON.stringify(INITIAL_FILTER)
+		);
+
+		if (urlParams.get('enddate')) {
+			initialFilter.activityPeriod.dates.endDate = urlParams.get(
+				'enddate'
+			)!;
+		}
+
+		if (urlParams.get('startdate')) {
+			initialFilter.activityPeriod.dates.startDate = urlParams.get(
+				'startdate'
+			)!;
+		}
+
+		if (urlParams.getAll('partner').length) {
+			initialFilter.partner.value = urlParams.getAll('partner')!;
+		}
+
+		if (urlParams.get('searchterm')) {
+			initialFilter.searchTerm = urlParams.get('searchterm')!;
+		}
+
+		if (urlParams.getAll('status').length) {
+			initialFilter.status.value = urlParams.getAll('status')!;
+		}
+
+		return initialFilter;
+	});
```

Comment:
Maybe you can create a callback for this

Reply:
Couldn't think in a efficient way to do this.